### PR TITLE
Postional arguments need to be quoted

### PR DESCRIPTION
**Pull request by lakshmi-kannan**
st2tests PR: https://github.com/StackStorm/st2tests/pull/26

### CODE REVIEW

File: st2common/st2common/util/action_db.py [hunk 1]
```diff
@@ -215,7 +215,6 @@ def get_args(action_parameters, action_db):
     for _, arg in six.iteritems(position_args_dict):
         positional_args.append(str(action_parameters.get(arg)))
         positional_args_keys.add(arg)
-    positional_args = ' '.join(positional_args)  # convert to string.
```
Comment (lakshmi-kannan) [inline st2common/st2common/util/action_db.py]: We were converting pos args to a string here which is not right. We should pass it as a list and let ssh runners deal with it. I am also not happy that this method exists in this file. This is specific to script runners. Anyways, one step at a time.

File: st2common/tests/unit/test_shell_action_system_model.py [hunk 2]
```diff
@@ -158,7 +158,7 @@ def test_command_construction_with_parameters(self):
         kwargs['sudo'] = False
         kwargs['user'] = LOGGED_USER_USERNAME
         kwargs['named_args'] = {'key1': 'value1', 'key2': 'value2'}
-        kwargs['positional_args'] = 'ein zwei drei'
+        kwargs['positional_args'] = ['ein', 'zwei', 'drei']
```
Comment (lakshmi-kannan) [inline st2common/tests/unit/test_shell_action_system_model.py]: These list changes are fairly easy to understand why I did it.

File: st2common/tests/unit/test_paramiko_script_action_model.py [hunk 3]
```diff
@@ -37,10 +37,11 @@ def test_get_command_string(self):
                                                    parallel=True,
                                                    sudo=False,
                                                    timeout=60)
-        expected = '/tmp/remote_script.sh song=\'b s\' "taylor swift"'
+        expected = '/tmp/remote_script.sh song=\'b s\' \'taylor swift\''
```
Comment (lakshmi-kannan) [inline st2common/tests/unit/test_paramiko_script_action_model.py]: This seems fine.

File: st2common/tests/unit/test_paramiko_script_action_model.py [hunk 4]
```diff
@@ -37,10 +37,11 @@ def test_get_command_string(self):
...
         self.assertEqual(script_action.get_full_command_string(), expected)
 
         # Test with sudo
         script_action.sudo = True
-        ex = 'sudo -E -- bash -c \'/tmp/remote_script.sh song=\'"\'"\'b s\'"\'"\' "taylor swift"\''
+        ex = 'sudo -E -- bash -c ' + \
+             '\'/tmp/remote_script.sh song=\'"\'"\'b s\'"\'"\' \'"\'"\'taylor swift\'"\'"\'\''
```
Comment (lakshmi-kannan) [inline st2common/tests/unit/test_paramiko_script_action_model.py]: Kinda skeptic whether this is fine. We need to write a sudo action that actually takes params with spaces (both positional and named args) and validate that args are passed in correctly. I'd be happy if we added this test in st2tests. cc: m4dcoder.

File: st2common/st2common/util/action_db.py (hunk 1 above)
Comment (Kami) [inline st2common/st2common/util/action_db.py]: Agreed.

File: st2common/st2common/models/system/action.py [hunk 5]
```diff
@@ -206,7 +206,8 @@ def _get_script_arguments(self, named_args=None, positional_args=None):
 
         # add the positional args
         if positional_args:
-            command_parts.append(positional_args)
+            pos_args_string = ' '.join(map(quote_unix, positional_args))
```
Comment (Kami) [inline st2common/st2common/models/system/action.py]: Any particular reason why you used map over a list comprehension?

In any case, LGTM.

Comment (lakshmi-kannan) [inline st2common/st2common/models/system/action.py]: I should really use list comprehension. I'll make that change.

File: st2common/tests/unit/test_paramiko_script_action_model.py (hunk 4 above)
Comment (lakshmi-kannan) [inline st2common/tests/unit/test_paramiko_script_action_model.py]: Verified behavior with st2tests PR.

**Comment by m4dcoder** [comment]
LGTM. Thank You!

**Pull request merged**